executor/executor.go

```diff
@@ -122,22 +143,92 @@ func (e *Executor) Run(conflicts state.Keys, f func() error) {
 	e.tasks[id] = t
 
 	// Record dependencies
-	for k := range conflicts {
-		latest, ok := e.edges[k]
-		if ok {
-			lt := e.tasks[latest]
-			if !lt.executed {
-				if t.dependencies == nil {
-					t.dependencies = set.NewSet[int](defaultSetSize)
+	for k, v := range conflicts {
+		// Get last blocked transaction
+		parent, exists := e.edges[k]
+		if !exists {
+			concurrentReads := set.NewSet[int](defaultSetSize)
+			if v == state.Read {
+				// Record any reads on the same key to help
+				// with updating dependencies properly
+				concurrentReads.Add(id)
+			}
+			// Key doesn't exist, so we add it to our edge map
+			e.edges[k] = &KeyData{Parent: id, Permissions: v, ConcurrentReads: concurrentReads}
```

Comment:
Don't love `Parent` here. Not sure what to replace it with but it is really `LastWriteAllocateOrRead` lol

Reply:
Maybe just `ID`?